internal/resources/cluster/tkgvsphere/resource_tkgvsphere.go

```diff
@@ -460,15 +460,19 @@ func flattenTKGVsphereTopology(topology *tkgvspheremodel.VmwareTanzuManageV1alph
 		return nil
 	}
 
-	flattenTopology[controlPlaneKey] = flattenTKGVsphereTopologyControlPlane(topology.ControlPlane)
+	if topology.ControlPlane != nil {
```

Comment:
this nil check should be handled in flattenTKGVsphereTopologyControlPlane function call